### PR TITLE
Add notebooks section on website

### PR DESCRIPTION
**Pull request by jackgerrits**
This PR adds support for rendering notebooks on the website. It automatically includes files from `/notebook` after they have a small amount of metadata added.

Once there are a sufficient number of notebooks opted in web can enable the Notebooks section in the navbar so users can find it.

To add a notebook to the website it should include a comment at the very top containing two yaml properties:
```markdown
<!--
tags: ["RAG"]
description: |
    Explore the use of AutoGen's RetrieveChat for tasks like code generation from docstrings, answering complex questions with human feedback, and exploiting features like Update Context, custom prompts, and few-shot learning.
-->
```
Screenshot 2024-01-31 at 7 08 03 PM
Screenshot 2024-01-31 at 7 07 59 PM

Screenshot 2024-01-31 at 7 08 48 PM

## Related issue number

#1339

### CODE REVIEW

**Comment by gagb** [comment]
Fantastic PR Jack-- it preserves the current dev flow for adding new notebooks and improves the website!

**Review by ekzhu — APPROVED** [review]
I really like this PR. Great work!

File: notebook/agentchat_function_call_async.ipynb [hunk 1]
```diff
@@ -73,9 +50,9 @@
     "import autogen\n",
     "from autogen.cache import Cache\n",
     "\n",
+    "# Exclude Azure OpenAI endpoints because some endpoints don't support functions yet. Remove the filter dict if they do.\n",
```
Comment (sonichi) [inline notebook/agentchat_function_call_async.ipynb:53]: This comment is outdated and should be removed.

**Review by sonichi — COMMENTED** [review]
One minor issue.

**Pull request merged**